static/js/stream_list.js

```diff
@@ -422,6 +452,16 @@ export function update_streams_sidebar(force_rerender) {
     update_stream_sidebar_for_narrow(filter);
 }
 
+export function update_dom_unread_counts_visibility() {
+    for (const [stream_id, stream] of stream_sidebar.rows.entries()) {
+        update_count_in_dom(
+            stream.get_li(),
+            unread.num_unread_for_stream(stream_id),
+            unread.stream_has_any_unread_mentions(stream_id),
+        );
+    }
+}
```

Comment:
Is this preferable to just calling `update_streams_sidebar(true)` from the server events code?

Reply:
Actually `update_stream_sidebar` never calls `update_read_counts` method on stream_rows even with `force_rerender=true`,
hence I made this separate function for exactly that 🙂

Reply:
OK but this function is explicitly document as something not to call in a loop:

```
export function stream_has_any_unread_mentions(stream_id) {                               
    // This function is somewhat inefficient and thus should not be                       
    // called in loops, since runs in O(total unread mentions) time.                      
    const streams_with_mentions = unread_topic_counter.get_streams_with_unread_mentions();
    return streams_with_mentions.has(stream_id);                                          
}                                                                                         
```

I'm not sure what the best solution is, but we should either do a single `get_counts` call or do something else.

Reply:
@timabbott I have changed some logic it is much more efficient now we won't need to call `stream_has_unread_mentions` in a loop! Please check out the latest commit, Thanks!